### DISCUSSION TITLE
Why does sessionmaker class use snake_case and it's instances use CamelCase?

### DISCUSSION THREAD

**Discussion by rrupy** [sqlalchemy/sqlalchemy#11073] [Usage Questions] [answered]
I think such naming contradicts PEP8 style

**Comment by zzzeek** [comment, marked as answer]
`Session` is a class, a factory for `Session` instances.   the sessionmaker is also a factory for`Session` instances, so to emphasize its interchangeability with the`Session` class, that's the naming convention in some of the docs right now.  a lot of people call it `maker()` instead

**Reply by rrupy** [reply]
what about sessionmaker instead of SessionMaker?

**Reply by zzzeek** [reply]
sessionmaker was originally a function

**Reply by rrupy** [reply]
historical reasons?:)

**Reply by zzzeek** [reply]
it also should be thought of as a function, it really shouldn't matter.  there's lots of functions that produce objects in SQLAlchemy and in Python std lib.     There could very well someday be a reason that "sessionmaker" is a constructor function again instead of a class.  we shouldnt have to change the name every time we change the implementation.   since you quote pep8, the first section after the intro is literally titled [A Foolish Consistency is the Hobgoblin of Little Minds](https://peps.python.org/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds)  and the primary phrase in that section is **In particular: do not break backwards compatibility just to comply with this PEP!**
